src/Aspire.Hosting/Health/ResourceHealthCheckService.cs

```diff
@@ -327,21 +342,26 @@ internal async Task<bool> DelayAsync(ResourceEvent? currentEvent, TimeSpan delay
                     _logger.LogTrace("Health monitoring delay interrupted for resource '{Resource}'.", _resourceName);
                     return true;
                 }
+                if (_delayCts == null || !_delayCts.TryReset())
+                {
+                    _delayCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                }
                 _delayInterruptTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
             }
 
-            try
-            {
-                await _delayInterruptTcs.Task.WaitAsync(delay, cancellationToken).ConfigureAwait(false);
+            var completedTask = await Task.WhenAny(Task.Delay(delay, _delayCts.Token), _delayInterruptTcs.Task).ConfigureAwait(false);
```

Comment:
@stephentoub it woud be great to have a non-throwing version of WaitAsync

Reply:
Old reliable > new hotness 😋

Reply:
There is:
await task.WaitAsync(timeout).ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing);